crates/xline/src/server/command.rs

```diff
@@ -337,6 +334,83 @@ impl CommandExecutor {
     }
 }
 
+/// After Sync Result
+type AfterSyncResult = Result<AfterSyncOk<Command>, <Command as CurpCommand>::Error>;
+
+/// Collection of after sync results
+struct ASResultStates<'a> {
+    /// After sync cmds
+    cmds: Vec<AfterSyncCmd<'a, Command>>,
+    /// After sync results
+    results: Vec<Option<AfterSyncResult>>,
+}
+
+impl<'a> ASResultStates<'a> {
+    /// Creates a new [`ASResultStates`].
+    fn new(cmds: Vec<AfterSyncCmd<'a, Command>>) -> Self {
+        Self {
+            results: iter::repeat_with(|| None::<AfterSyncResult>)
+                .take(cmds.len())
+                .collect(),
+            cmds,
+        }
```

Comment:
I don't know if the `cmds` and `results` are one-to-one relationship. 
If the answer is "yes", why not use a `Vec<(AfterSyncCmd<'a, Command>, Option<AfterSyncResult>)>` instead?

Reply:
Yes, they are one-to-one relationship. I have write a new version of `ASResultStates`